sdk/identity/azure-identity/test/ut/azure_cli_credential_test.cpp

```diff
@@ -344,6 +344,50 @@ TEST(AzureCliCredential, UnsafeChars)
   }
 }
 
+TEST(AzureCliCredential, UnsafeCharsScopeAndTenantId)
+{
+  std::string const ValidScopeButNotTenantId = "string:withcolon";
+
+  {
+    AzureCliCredentialOptions options;
+    options.TenantId = "01234567-89AB-CDEF-0123-456789ABCDEF";
+    options.TenantId += ValidScopeButNotTenantId;
+    AzureCliCredential azCliCred(options);
+
+    TokenRequestContext trc;
+    trc.Scopes.push_back(std::string("https://storage.azure.com/.default"));
+    EXPECT_THROW(static_cast<void>(azCliCred.GetToken(trc, {})), AuthenticationException);
+
+    try
+    {
+      auto const token = azCliCred.GetToken(trc, {});
+    }
+    catch (AuthenticationException const& e)
+    {
+      EXPECT_TRUE(std::string(e.what()).find("Unsafe") != std::string::npos) << e.what();
```

Comment:
I'd go and hardcode the expected exception message, if not the whole message - then the larger part. After all, error messages, and diagnosability is also a feature we deliver, therefore it should be tested, and it is totally normal to expect a certain text here. It is the Identity library code that makes the error message, so it is under our control. If we change the error message, we update the test.

Reply:
We should come up with guidelines on this. Testing for exact error messages adds brittleness to unit testing (unnecessary tight coupling), and historically (from my .NET experience), we'd avoid doing such things.